src/prop/bvminisat/bvminisat.cpp

```diff
@@ -278,5 +276,24 @@ void BVMinisatSatSolver::Statistics::init(BVMinisat::SimpSolver* minisat){
   d_statEliminatedVars.set(minisat->eliminated_vars);
 }
 
+void BVMinisatSatSolver::Statistics::deinit()
```

Comment:
Why a `deinit()` method instead of doing the reset in `~Statistics`?

Reply:
I found it way easier to do this manually instead of relying in destruction order, in particular in this situation. This (or rather something equivalent) is done in `~Statistics` anyway. The issue here is that the member (d_statistics) is destructed after the manual destructor has run, which destroys d_minisat manually.